actionpack/lib/action_controller/metal/params_wrapper.rb

```diff
@@ -91,7 +91,7 @@ def initialize(name, format, include, exclude, klass, model) # :nodoc:
       end
 
       def model
-        super || synchronize { super || self.model = _default_wrap_model }
+        super || self.model = _default_wrap_model
```

Comment:
I think what is important to synchronize is the assignment to `self.model`. There is nothing in the `_default_wrap_model` what seems to require lock.

Reply:
True, and in that case my patch is invalid. The deadlock is here: https://github.com/sergiogomez/rails/blob/af7f7c9beff52aceb2f6ab7368e50e6d7820d411/actionpack/lib/action_controller/metal/params_wrapper.rb#L157

Reply:
Maybe what we need to do is make sure the mutex we are using is loading aware. We could try to use this class. https://github.com/rails/rails/blob/master/activesupport/lib/active_support/concurrency/load_interlock_aware_monitor.rb